### ISSUE TITLE
[Android] [CLI] Unable to use CLI to generate android package with dashes in the name.

### ISSUE THREAD

**Issue by geof90** [facebook/react-native#4216]
The android project generator from the CLI, e.g `react-native android`, fails if the name specified in package.json contains dashes, eg "my-react-app". Instead of rejecting names with dashes, can we strip out the dashes in dash-cased names, e.g. "my-react-app" -> "com.myreactapp" when generating the java package? This is really helpful because NPM modules are generally dash-cased, while Java packages are usually all lower cased.

**Comment by mkonicek** [comment]
Sounds like a good idea!

**Comment by mkonicek** [comment]
@facebook-github-bot bugfix

**Comment by farwayer** [comment]
I just ran into this problem with `react-native upgrade`...

**Comment by anirudh24seven** [comment]
This issue is also dependent on #7127 to be closed

**Comment by lacker** [comment]
This issue has been inactive for a while, and it seems like a "nice to have" but noncritical since there's a clear workaround. I'm going to close this issue but if someone wants to fix this up and send a PR that would be great!
